samples/create-buffers-around-points/src/main/java/com/esri/arcgismaps/sample/createbuffersaroundpoints/components/CreateBuffersAroundPointsViewModel.kt

```diff
@@ -0,0 +1,291 @@
+                return@launch
+            }
+
+            // Store the tapped point and request input dialog
+            lastTappedPoint = mapPoint
+            _isInputDialogVisible.value = true
+            _statusText.value = "Enter buffer radius (miles) for the tapped location"
+        }
+    }
+
+    /**
+     * Submit the radius (in miles) provided by the user. Converts miles to the map's linear units
+     * (the state plane spatial reference uses US feet) then creates the buffer and updates the overlays.
+     */
+    fun submitRadiusMiles(radiusMiles: Double) {
+        viewModelScope.launch {
+            val point = lastTappedPoint
+            if (point == null) {
+                messageDialogVM.showMessageDialog("Internal error: missing tapped point")
+                _isInputDialogVisible.value = false
+                return@launch
+            }
+
+            if (radiusMiles <= 0.0) {
+                messageDialogVM.showMessageDialog("Please enter a value greater than 0")
+                return@launch
+            }
+
+            // Convert miles to feet (1 mile = 5280 feet). The state plane uses US feet for this sample.
+            val radiusFeet = radiusMiles * 5280.0
+
+            // Add to internal list and draw
+            bufferPoints.add(point to radiusFeet)
+
+            // Add tap point graphic
+            tappedPointsGraphicsOverlay.graphics.add(Graphic(geometry = point))
+
+            // Redraw buffers using current union setting
+            drawBuffers(unioned = _shouldUnion.value)
+
+            _isInputDialogVisible.value = false
+            _statusText.value = "Buffer created. Tap to add more points or press Clear."
+        }
+    }
+
+    /**
+     * Draws buffers for all stored bufferPoints. If unioned is true, attempt to union buffers
+     * into a single geometry before displaying.
+     */
+    fun drawBuffers(unioned: Boolean) {
+        viewModelScope.launch {
+            // Clear existing buffer graphics
+            bufferGraphicsOverlay.graphics.clear()
+
+            if (bufferPoints.isEmpty()) {
+                _statusText.value = "Add a point to draw the buffers."
+                return@launch
+            }
+
+            // Create buffer geometries for each point
+            val polygons = bufferPoints.mapNotNull { (pt, radius) ->
+                GeometryEngine.bufferOrNull(pt, radius)
+            }
+
+            if (polygons.isEmpty()) {
+                messageDialogVM.showMessageDialog("Error creating buffer geometries")
+                return@launch
+            }
+
+            if (unioned) {
+                // Union the polygons into a single geometry (may return a Polygon or Multipart geometry)
+                val unionedGeometry = GeometryEngine.unionOrNull(polygons)
+                if (unionedGeometry != null) {
+                    bufferGraphicsOverlay.graphics.add(Graphic(geometry = unionedGeometry))
```

Comment:
Could just be an implicit check here since we already check for `polygons.isEmpty()` above: 

```suggestion
                // Union the polygons into a single geometry (may return a Polygon or Multipart geometry)
                GeometryEngine.unionOrNull(polygons)?.let { unionedGeometry ->
                    bufferGraphicsOverlay.graphics.add(Graphic(geometry = unionedGeometry))
                }
```